code/tasks/SiteTreeFullBuildEngine.php

```diff
@@ -96,14 +96,33 @@ protected function runFrom($start) {
 		$pages = $this->getAllLivePages()->sort('ID')->limit($chunkSize, $start);
 		$count = 0;
 
+		$arrNonCachedPages = array();
+
 		// Collect all URLs into the queue
 		foreach($pages as $page) {
+
+			if (is_callable(array($page, 'canCache')) && !$page->canCache()) {
+				$arrNonCachedPages[] = $page;
+				continue;
+			}
+
 			if (is_callable(array($page, 'urlsToCache'))) {
 				$this->getUrlArrayObject()->addUrlsOnBehalf($page->urlsToCache(), $page);
 				$count++;
 			}
 		}
 
+		$publisher = singleton('SiteTree')->getExtensionInstance('FilesystemPublisher');
+		if ($publisher) {
+			foreach ($arrNonCachedPages as $page) {
+				$pathsArray = $publisher->urlsToPaths(array($page->Link()));
+				$filePath = $publisher->getDestDir() . '/' . array_shift($pathsArray);
+				@unlink($filePath);
```

Comment:
Sorry was a cut and pasted from this module... they must have been removed recently